src/ZenodoFile.ts

```diff
@@ -0,0 +1,52 @@
+import { z } from 'zod';
+
+import type { Zenodo } from './Zenodo';
+
+// Define the Zod schema
```

Comment:
@targos I think this is the least issue with this PR. I have the impression to write 3 times the same code. Is this really the correct way ?

I want to be sure before doing the same for Deposition that is 10 times more complex

Reply:
I would just put the value coming from Zenodo in a property: `this.xxx = validatedFile`.